pkg/resource/deployment_test.go

```diff
@@ -0,0 +1,220 @@
+				"credentials": []byte("updated creds"),
+			}),
+			validate: func(t *testing.T, origHash string, dep *appsapi.Deployment) {
+				currentHash := dep.Annotations[defaults.ChecksumOperatorAnnotation]
+				if origHash == currentHash {
+					t.Errorf("Hash unexpectedly didn't change from %s", origHash)
+				}
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			annotatedNamespace := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: defaults.ImageRegistryOperatorNamespace,
+					Annotations: map[string]string{
+						defaults.SupplementalGroupsAnnotation: "1/2",
+					},
+				},
+			}
+
+			imageRegistryCertificatesConfigMap := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      defaults.ImageRegistryCertificatesName,
+					Namespace: defaults.ImageRegistryOperatorNamespace,
+				},
+			}
+
+			kubeClient := fake.NewSimpleClientset(test.origSecret, annotatedNamespace, imageRegistryCertificatesConfigMap)
+			kubeInformer := kubeinformers.NewSharedInformerFactory(kubeClient, 0)
+
+			configClient := fakeconfig.NewSimpleClientset()
+			configInformer := configinformers.NewSharedInformerFactory(configClient, 0)
+
+			cmLister := kubeInformer.Core().V1().ConfigMaps().Lister().ConfigMaps(defaults.ImageRegistryOperatorNamespace)
+			secretLister := kubeInformer.Core().V1().Secrets().Lister().Secrets(defaults.ImageRegistryOperatorNamespace)
+
+			proxyLister := configInformer.Config().V1().Proxies().Lister()
+
+			kubeInformer.Start(ctx.Done())
+			configInformer.Start(ctx.Done())
+
+			gd := &generatorDeployment{
+				driver:          &testDriver{},
+				coreClient:      kubeClient.CoreV1(),
+				proxyLister:     proxyLister,
+				cr:              &imageregistryv1.Config{},
+				configMapLister: cmLister,
+				secretLister:    secretLister,
+			}
+
+			cache.WaitForCacheSync(ctx.Done(), kubeInformer.Core().V1().ConfigMaps().Informer().HasSynced)
+			cache.WaitForCacheSync(ctx.Done(), kubeInformer.Core().V1().Secrets().Informer().HasSynced)
+
+			// Generate a base deployment
+			obj, err := gd.expected()
+			if err != nil {
+				t.Fatalf("unexpected error: %s", err)
+			}
+
+			generatedDeployment, ok := obj.(*appsapi.Deployment)
+			if !ok {
+				t.Fatalf("failed to convert to Deployment")
+			}
+
+			origHash := generatedDeployment.Annotations[defaults.ChecksumOperatorAnnotation]
+
+			if test.updatedSecret != nil {
+				_, err = kubeClient.CoreV1().Secrets(defaults.ImageRegistryOperatorNamespace).Update(ctx, test.updatedSecret, metav1.UpdateOptions{})
+				if err != nil {
+					t.Fatalf("failed to update secret: %s", err)
+				}
+
+				cache.WaitForCacheSync(ctx.Done(), kubeInformer.Core().V1().Secrets().Informer().HasSynced)
+				// Fixme: why is this wait necessary?
+				if err := waitForUpdatedSecret(ctx, kubeClient, test.updatedSecret.Data); err != nil {
```

Comment:
nit: this should not be necessary, does the test fail if you remove this wait?

Reply:
It's been a while since I touched this, but I believe I added it because I was running into issue otherwise.

Reply:
Well, this is not a no go condition, let's just move this forward.